cascade.py

```
@@ -1,146 +1,295 @@
+import asyncio
+import signal
+import threading
+import config
+import logging
+import traceback
 import numpy as np
+import xml.etree.ElementTree as ET
+import multiprocessing
+from contextlib import contextmanager, suppress
+from logging_setup import setup_logging
 from pathlib import Path
```

Ruff (0.12.2)

`pathlib.Path` imported but unused

Remove unused import: `pathlib.Path`

(F401)